### PR TITLE
Added watch later functionality

### PR DESCRIPTION
**Pull request by vijaytembugade**
- Added  add to watch-later functionality
- watch later context 
- watch later screens

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ import PlayList from "./Pages/PlayList/PlayList";
 import PlayListDetails from "./Pages/PlayListDetails/PlayListDetails";
 import { ScrollToTop } from "./Components/ScrollToTop/ScrollToTop";
```
Comment (t007rushi) [inline src/App.js:14]: new thing seen

File: src/Components/VideoCard/VideoCard.js [hunk 2]
```diff
@@ -12,9 +17,14 @@ const VideoCard = ({ video }) => {
   } = useAuth();
 
   const { likedVideos, setLikedVideos } = useLikedVideo();
+  const { watchLaterVideos, setWatchLaterVideos } = useWatchLater();
+
   const isLikedVideo = likedVideos.find((likedVideo) =>
     likedVideo._id === video._id ? true : false
```
Comment (t007rushi) [inline src/Components/VideoCard/VideoCard.js:23]: is there need of explicit check?

File: src/Components/VideoCard/VideoCard.js [hunk 3]
```diff
@@ -75,6 +84,47 @@ const VideoCard = ({ video }) => {
...
+      toast.error(error.message);
+    }
+  }
+
+  async function handleRemoveFromWatchLater(e) {
+    e.stopPropagation();
+    try {
+      if (isLoggedIn) {
+        const response = await removeFromWatchLaterService(video._id, token);
+        console.log(response);
```
Comment (t007rushi) [inline src/Components/VideoCard/VideoCard.js:113]: remove console here

File: src/Components/VideoCard/VideoCard.js [hunk 4]
```diff
@@ -75,6 +84,47 @@ const VideoCard = ({ video }) => {
       toast.error(error.message);
     }
   }
+
+  async function handleAddToWatchLater(e) {
+    e.stopPropagation();
+    try {
+      if (isLoggedIn) {
+        const response = await addToWatchLaterService(video, token);
```
Comment (t007rushi) [inline src/Components/VideoCard/VideoCard.js:92]: you can destructure {data} here itself

File: src/Components/VideoDescription/VideoDescription.js [hunk 5]
```diff
@@ -76,6 +87,47 @@ const VideoDescription = ({ video }) => {
...
+      toast.error(error.message);
+    }
+  }
+
+  async function handleRemoveFromWatchLater(e) {
+    e.stopPropagation();
+    try {
+      if (isLoggedIn) {
+        const response = await removeFromWatchLaterService(video._id, token);
+        console.log(response);
```
Comment (t007rushi) [inline src/Components/VideoDescription/VideoDescription.js:116]: same here

**Review by t007rushi — APPROVED** [review]
Look for minor left-outs in code else Great Lib. 
with Great Bg-image for login 
consistent in theme

File: src/Components/VideoCard/VideoCard.js (hunk 3 above)
Comment (horsemaker) [inline src/Components/VideoCard/VideoCard.js:113]: Do remove the `console.log()` here.

File: src/Components/VideoCard/VideoCard.js [hunk 6]
```diff
@@ -75,6 +84,47 @@ const VideoCard = ({ video }) => {
       toast.error(error.message);
     }
   }
+
+  async function handleAddToWatchLater(e) {
+    e.stopPropagation();
```
Comment (horsemaker) [inline src/Components/VideoCard/VideoCard.js:89]: Good use of `e.stopPropagation()`

File: src/Components/VideoDescription/VideoDescription.js (hunk 5 above)
Comment (horsemaker) [inline src/Components/VideoDescription/VideoDescription.js:116]: Remove the `console.log()` here

**Review by horsemaker — APPROVED** [review]

**Pull request merged**